src/Compilers/CSharp/Portable/Binder/LocalScopeBinder.cs

```diff
@@ -181,18 +181,26 @@ protected ImmutableArray<LocalSymbol> BuildLocals(SyntaxList<StatementSyntax> st
                         }
                     case SyntaxKind.LocalDeclarationStatement:
                         {
+                            enclosingBinder = enclosingBinder.GetBinder(innerStatement) ?? enclosingBinder;
                             var decl = (LocalDeclarationStatementSyntax)innerStatement;
                             LocalDeclarationKind kind = decl.IsConst ? LocalDeclarationKind.Constant : LocalDeclarationKind.RegularVariable;
                             foreach (var vdecl in decl.Declaration.Variables)
                             {
-                                var localSymbol = MakeLocal(decl.Declaration, vdecl, kind);
+                                var localSymbol = MakeLocal(decl.Declaration, vdecl, kind, enclosingBinder);
                                 locals.Add(localSymbol);
+
+                                var value = vdecl.Initializer?.Value;
+                                if (value != null)
```

Comment:
`FindExpressionVariables` already exits early if its third parameter is `null`. Consider removing conditional?